scandium-core/src/test/java/org/eclipse/californium/scandium/DTLSConnectorTest.java

```diff
@@ -808,24 +779,23 @@ public void testConnectorSendsHelloVerifyRequestWithoutCreatingSession() throws
 		final List<Record> receivedRecords = new ArrayList<>();
 		InetSocketAddress endpoint = new InetSocketAddress(12000);
 
-		DataHandler handler = new DataHandler() {
-
+		LatchDecrementingDataHandler handler = new LatchDecrementingDataHandler(latch) {
 			@Override
-			public void handleData(byte[] data) {
-				receivedRecords.addAll(Record.fromByteArray(data, serverEndpoint, null)); 
-				latch.countDown();
+			public boolean process(InetSocketAddress endpoint, final byte[] data) {
```

Comment:
any reason to make `data` final ?

Reply:
"code completion" on the old definition, which uses that final.
I removed the final on all occurrences.
It's completely removed with a later clean up.